### DISCUSSION TITLE
API GET tickets with filter

### DISCUSSION THREAD

**Discussion by rodrigolfh** [django-helpdesk/django-helpdesk#1214] [Q&A] [answered]
How can I use the /api/tickets GET endpoint to filter tickets by submitter_email?

Thanks in advance

**Comment by Benbb96** [comment]
Hi @rodrigolfh, the tickets endpoint currently has only one filter possible (status). But the submitter_email can easily be added. Maybe by using [django-filter](https://django-filter.readthedocs.io/en/latest/index.html).

I can make a PR for that 😃

**Reply by uhurusurfa** [reply]
@Benbb96 If you do raise a PR for it, please increment the version number in setup.py from 1.2.0 to 1.3.0 so we can make a new release with all the enhancements and fixes that have gone in recently.

**Comment by rodrigolfh** [comment, marked as answer]
I finally decided to go with a 'run' filter. All citizens here in Chile have a 'run', an integer identifier. It is a 7 to 9 digit number along with a verifying digit which results from some arithmetic from the previous integer. I only used the main number which is unique to every citizen.

1. I modified the Ticket model adding an attribute:

```
    submitter_run = models.IntegerField(
        _('Submitter RUN'),
        blank=True,
        null=True,
        help_text=_('Ingresar Run sin DV'),
    )
```


2. Made and ran the migrations.

3. Modified the method get_queryset() in the TicketViewSet Class:

```
def get_queryset(self):
        tickets = Ticket.objects.all()

        # filter by status
        status = self.request.query_params.get('status', None)
        
        # get run
        run = self.request.query_params.get('run', None)


        if status:
            statuses = status.split(',') if status else []
            status_choices = helpdesk_settings.TICKET_STATUS_CHOICES
            number_statuses = []
            for status in statuses:
                for choice in status_choices:
                    if str(choice[0]) == status:
                        number_statuses.append(choice[0])
            if number_statuses:
                tickets = tickets.filter(status__in=number_statuses)                

        # filter again by run 
        if run:
            tickets = tickets.filter(submitter_run=run)
        
        return tickets
```
